app.rb

```diff
@@ -57,54 +73,151 @@ def create_book
     book = Book.new(title, author)
     @books.push(book)
     puts "Book '#{title}' created successfully"
+    save_books_to_json
   end
 
   def create_rental
     puts 'Select a book from the following list by number'
-    @books.each_with_index do |book, index|
-      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
-    end
-    book_index = gets.chomp.to_i
+    list_books
+    book_index = gets.chomp.to_i - 1
+
     puts 'Select a person from the following list by number (not id)'
-    @people.each_with_index do |person, index|
-      type = person.instance_of?(Student) ? 'Student' : 'Teacher'
-      puts "#{index}) [#{type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
-    end
-    person_index = gets.chomp.to_i
+    list_people
+    person_index = gets.chomp.to_i - 1
+
     print 'Date: '
     date = gets.chomp
-    @rentals.push(Rental.new(date, @books[book_index], @people[person_index]))
+
+    rental = Rental.new(date, @books[book_index], @people[person_index])
+    @rentals.push(rental)
     puts 'Rental created successfully'
+    save_rentals_to_json
   end
 
+  #------------- lists
+
   def list_books
+    puts 'List of Books:'
     @books.each_with_index do |book, index|
-      puts "#{index} - Title: #{book.title}, Author: #{book.author}"
+      puts "#{index + 1}. Title: #{book.title}, Author: #{book.author}"
     end
   end

```

Comment:
- [ ] After adding books to the app json files, I can list them without any errors, but when I quit the app restart it, and try to list the same books again, I get errors and the books are not listed. Kindly fix that.

  ![image](https://github.com/rodrigopenayo98/OOP-school-librery/assets/54848363/244066d7-747f-4c5f-8dea-ed11ed81ff45)

Reply:
well resolved! ✔️